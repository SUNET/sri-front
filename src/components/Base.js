import React from "react";
import { Route, Switch } from "react-router-dom";

import SearchContainer from "../containers/Search";
import { ContactDetails } from "./Contact";

class Base extends React.Component {
    render() {
        return (
            <main className="container">
                <Switch>
                    <Route path="/contacts" component={SearchContainer} />
                    <Route
                        path="/contact/:contactId"
                        component={ContactDetails}
                    />
                    <Route path="/roles" component={SearchContainer} />
                </Switch>
            </main>
        );
    }
}

export default Base;
