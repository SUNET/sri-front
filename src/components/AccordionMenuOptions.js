import React from "react";
import { Nav } from "react-bootstrap";
import { NavLink } from "react-router-dom";
import ToggleSection, { ToggleHeading, TogglePanel } from "../components/ToggleSection";

import "../style/AccordionMenuOptions.scss";

class AccordionMenuOptions extends React.Component {
    renderHeader(option, index) {
        return (
            <ToggleHeading key={`${index}`}>
                <div className="accordion-menu-options__header">{option.header}</div>
            </ToggleHeading>
        );
    }
    renderCollapseList(subOptions, optionIndex) {
        return (
            <TogglePanel>
                {subOptions.map((subOption, subOpIndex) => {
                    return (
                        <div key={`option_menu-${optionIndex}-${subOpIndex}`} className="accordion-menu-options__link">
                            <Nav.Link
                                as={NavLink}
                                activeClassName="active"
                                to={subOption.route}
                                onClick={() => this.props.onSelectedOption()}
                            >
                                <i className={`accordion-menu-options__link__icon ${subOption.iconClass}`}></i>
                                <span className="accordion-menu-options__link__name">{subOption.name}</span>
                            </Nav.Link>
                        </div>
                    );
                })}
            </TogglePanel>
        );
    }
    render() {
        const { data } = this.props;

        return (
            <div className="accordion-menu-options">
                {data.map((option, optionIndex) => {
                    return (
                        <ToggleSection key={`option_menu-${optionIndex}`}>
                            {this.renderHeader(option, optionIndex)}
                            {this.renderCollapseList(option.subOptions, optionIndex)}
                        </ToggleSection>
                    );
                })}
            </div>
        );
    }
}

export default AccordionMenuOptions;
