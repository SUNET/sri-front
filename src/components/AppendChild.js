import React from 'react';
import { Field } from 'redux-form';
import { withTranslation } from 'react-i18next';
import FieldInput from './FieldInput';

class AppendChild extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      name: '',
      fields: [{ name: '' }],
    };
  }

  componentWillMount() {}

  handleFieldNameChange = (index) => (event) => {
    const newShareholders = this.state.fields.map((field, sindex) => {
      if (index !== sindex) return field;
      return { ...field, name: event.target.value };
    });

    this.setState({ fields: newShareholders });
  };

  handleAddFild = (event) => {
    event.preventDefault();
    this.setState({
      fields: this.state.fields.concat([{ name: '' }]),
    });
  };

  render() {
    let { t } = this.props;
    return (
      <div>
        {this.state.fields.map((field, index) => (
          <Field
            component={FieldInput}
            type="text"
            placeholder={`${field.name}${index + 1}`}
            value={field.name}
            onChange={this.handleFieldNameChange(index)}
          />
        ))}
        <button className="btn btn-add outline" onClick={(e) => this.handleAddField(e)}>
          <span>{t('actions/add-new')}</span>
        </button>
      </div>
    );
  }
}

export default withTranslation()(AppendChild);
