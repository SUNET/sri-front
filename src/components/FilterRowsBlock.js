import React from "react";
import { withTranslation } from "react-i18next";
import { Row, Col } from "react-bootstrap";
import Filter from "./Filter";
import OrderBy from "./OrderBy";
import RangeDayPicker from "./RangeDayPicker";

import "../style/Footer.scss";

class FilterRowsBlock extends React.Component {
    renderDateFilter() {
        const { t, changeFilterDateType, filterDateType, handleDateTo, handleDateFrom, handleResetDate } = this.props;
        return (
            <div className="data-filter-by-date">
                <div className="filter-date d-inline">
                    <div className="pretty p-default p-round">
                        <input
                            type="radio"
                            name="filterDateType"
                            checked={filterDateType === "created"}
                            value="created"
                            onChange={(e) => {
                                changeFilterDateType(e);
                            }}
                        />
                        <div className="state p-info-o">
                            <label>{t("filter.date.created")}</label>
                        </div>
                    </div>

                    <div className="pretty p-default p-round">
                        <input
                            type="radio"
                            name="filterDateType"
                            checked={filterDateType === "modified"}
                            value="modified"
                            onChange={(e) => {
                                changeFilterDateType(e);
                            }}
                        />
                        <div className="state p-info-o">
                            <label>{t("filter.date.updated")}</label>
                        </div>
                    </div>
                </div>
                <RangeDayPicker
                    dateTo={(event) => handleDateTo(event)}
                    dateFrom={(event) => handleDateFrom(event)}
                    resetDate={(event) => handleResetDate(event)}
                />
            </div>
        );
    }

    renderFilterByWord() {
        return <Filter changeFilter={this.props.handleOnChangeFilter} />;
    }
    renderOrderBy() {
        return <OrderBy changeOrderBy={this.props.handleOnChangeOrderBy} />;
    }
    renderFiltersBoxDesktop() {
        return (
            <Row>
                <Col>{this.renderDateFilter()}</Col>
                <Col className="text-right" sm={4}>
                    {this.renderFilterByWord()}
                    {this.renderOrderBy()}
                </Col>
            </Row>
        );
    }
    renderFiltersBoxMobile() {
        return (
            <div>
                <Col>
                    <Row className="justify-content-center">{this.renderFilterByWord()}</Row>
                    <hr />
                    <Row className="justify-content-center">{this.renderDateFilter()}</Row>
                    <hr />
                    <Row className="justify-content-center">{this.renderOrderBy()}</Row>
                    <hr />
                </Col>
            </div>
        );
    }
    render() {
        const { browserVersion } = this.props;
        return <div>{browserVersion ? this.renderFiltersBoxDesktop() : this.renderFiltersBoxMobile()}</div>;
    }
}

export default withTranslation()(FilterRowsBlock);
