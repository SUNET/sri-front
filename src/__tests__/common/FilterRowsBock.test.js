import React from 'react';
import { mount, shallow } from 'enzyme';
import { FilterRowsBlock } from '../../components/FilterRowsBlock';
import toJson from 'enzyme-to-json';
import * as deviceDetect from 'react-device-detect';

let FilterRowBlockWrapper, FilterRowBlockComponent;
jest.mock('react-i18next', () => ({
  withTranslation: () => (Component) => {
    Component.defaultProps = { ...Component.defaultProps, t: () => '' };
    return Component;
  },
}));

const defaultProps = {
  handleOnChangeFilter: jest.fn(),
  handleOnChangeOrderBy: jest.fn(),
  filterDateType: 'created', // "created || modified"
  handleDateTo: jest.fn(),
  handleDateFrom: jest.fn(),
  handleResetDate: jest.fn(),
  changeFilterDateType: jest.fn(),
  t: jest.fn((t) => t),
};

const CreateFilterRowBlockWrapper = (props) => {
  const allProps = { ...defaultProps, ...props };
  return (
    <div>
      <FilterRowsBlock {...allProps} />
    </div>
  );
};

const mountFilterRowBlockWrapper = (props) => mount(<CreateFilterRowBlockWrapper {...props} />);
beforeEach(() => {
  FilterRowBlockWrapper = mountFilterRowBlockWrapper();
  FilterRowBlockComponent = FilterRowBlockWrapper.find('FilterRowsBlock');
});
afterEach(() => {
  FilterRowBlockWrapper.unmount();
});

const getDateFilterCriteriaInputs = (checkedNameElement) => {
  const wrapperWithChecked = mountFilterRowBlockWrapper({ filterDateType: checkedNameElement });
  const createdEl = wrapperWithChecked.find('[data-name="filter-date-created"]');
  const updatedEl = wrapperWithChecked.find('[data-name="filter-date-updated"]');
  const inputCreated = createdEl.find('input');
  const inputUpdated = updatedEl.find('input');
  return {
    inputCreated,
    inputUpdated,
  };
};

describe('Filter Row Block Component', () => {
  it('Renders correctly Desktop Version', () => {
    deviceDetect.isBrowser = true;
    deviceDetect.isMobile = false;
    expect(toJson(FilterRowBlockWrapper)).toMatchSnapshot();
    expect(FilterRowBlockWrapper.find('.data-filter-by-date').length).toBe(1);
  });
  it('Renders correctly Mobile Version', () => {
    deviceDetect.isBrowser = false;
    deviceDetect.isMobile = true;
    expect(toJson(FilterRowBlockWrapper)).toMatchSnapshot();
    expect(FilterRowBlockWrapper.find('.data-filter-by-date').length).toBe(1);
  });
  describe('Created or Updated Option', () => {
    let filterDateBlock, createdElement, modifiedElement;
    beforeEach(() => {
      filterDateBlock = FilterRowBlockWrapper.find('.filter-date');
      createdElement = filterDateBlock.children('[data-name="filter-date-created"]');
      modifiedElement = filterDateBlock.children('[data-name="filter-date-updated"]');
    });

    it('Render Created or Updated filter date', () => {
      expect(filterDateBlock.children('.pretty.p-default.p-round')).toHaveLength(2);
    });
    it('Render Created Label', () => {
      const label = createdElement.find('label');
      expect(createdElement).toHaveLength(1);
      expect(label).toHaveLength(1);
      expect(label.text()).toEqual('filter.date.created');
    });
    it('Render Updated Label', () => {
      const label = modifiedElement.find('label');
      expect(modifiedElement).toHaveLength(1);
      expect(label).toHaveLength(1);
      expect(label.text()).toEqual('filter.date.updated');
    });
    it('Created is Checked', () => {
      const { inputCreated, inputUpdated } = getDateFilterCriteriaInputs('created');
      expect(inputCreated.getElement().props.checked).toEqual(true);
      expect(inputUpdated.getElement().props.checked).toEqual(false);
    });
    it('Updated is Checked', () => {
      const { inputCreated, inputUpdated } = getDateFilterCriteriaInputs('updated');
      expect(inputCreated.getElement().props.checked).toEqual(false);
      expect(inputUpdated.getElement().props.checked).toEqual(true);
    });
  });
  describe('Render RangeDayPicker', () => {
    it('Exists element', () => {
      expect(FilterRowBlockComponent.find('RangeDayPicker')).toHaveLength(1);
    });
  });
  describe('Render Filter By Words', () => {
    it('Exists element', () => {
      expect(FilterRowBlockComponent.find('Filter')).toHaveLength(1);
    });
  });
  describe('Render Order By', () => {
    it('Exists element', () => {
      expect(FilterRowBlockComponent.find('OrderBy')).toHaveLength(1);
    });
  });
});
