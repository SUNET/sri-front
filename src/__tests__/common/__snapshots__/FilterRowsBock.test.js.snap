// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filter Row Block Component Renders correctly Desktop Version 1`] = `
<CreateFilterRowBlockWrapper>
  <div>
    <FilterRowsBlock
      changeFilterDateType={[MockFunction]}
      filterDateType="created"
      handleDateFrom={[MockFunction]}
      handleDateTo={[MockFunction]}
      handleOnChangeFilter={[MockFunction]}
      handleOnChangeOrderBy={[MockFunction]}
      handleResetDate={[MockFunction]}
      t={
        [MockFunction] {
          "calls": Array [
            Array [
              "filter.date.created",
            ],
            Array [
              "filter.date.updated",
            ],
          ],
          "results": Array [
            Object {
              "type": "return",
              "value": "filter.date.created",
            },
            Object {
              "type": "return",
              "value": "filter.date.updated",
            },
          ],
        }
      }
    >
      <div>
        <Row
          noGutters={false}
        >
          <div
            className="row"
          >
            <Col
              lg={12}
              md={12}
              sm={12}
              xl={7}
              xs={12}
            >
              <div
                className="col-xl-7 col-lg-12 col-md-12 col-sm-12 col-12"
              >
                <div
                  className="data-filter-by-date"
                >
                  <div
                    className="filter-date d-inline"
                  >
                    <div
                      className="pretty p-default p-round"
                      data-name="filter-date-created"
                    >
                      <input
                        checked={true}
                        name="filterDateType"
                        onChange={[Function]}
                        type="radio"
                        value="created"
                      />
                      <div
                        className="state p-info-o"
                      >
                        <label>
                          filter.date.created
                        </label>
                      </div>
                    </div>
                    <div
                      className="pretty p-default p-round"
                      data-name="filter-date-modified"
                    >
                      <input
                        checked={false}
                        name="filterDateType"
                        onChange={[Function]}
                        type="radio"
                        value="modified"
                      />
                      <div
                        className="state p-info-o"
                      >
                        <label>
                          filter.date.updated
                        </label>
                      </div>
                    </div>
                  </div>
                  <RangeDayPicker
                    dateFrom={[Function]}
                    dateTo={[Function]}
                    resetDate={[Function]}
                    t={[Function]}
                  >
                    <div
                      className="range-day-picker"
                    >
                      <div
                        className="input-from-to input-from-to--from"
                      >
                        <label />
                        <DayPickerInput
                          classNames={
                            Object {
                              "container": "DayPickerInput",
                              "overlay": "DayPickerInput-Overlay",
                              "overlayWrapper": "DayPickerInput-OverlayWrapper",
                            }
                          }
                          clickUnselectsDay={false}
                          component="input"
                          dayPickerProps={
                            Object {
                              "disabledDays": Object {
                                "after": undefined,
                              },
                              "locale": "en",
                              "localeUtils": Object {
                                "formatDate": [Function],
                                "formatDay": [Function],
                                "formatMonthTitle": [Function],
                                "formatWeekdayLong": [Function],
                                "formatWeekdayShort": [Function],
                                "getFirstDayOfWeek": [Function],
                                "getMonths": [Function],
                                "parseDate": [Function],
                              },
                              "modifiers": Object {
                                "end": undefined,
                                "start": undefined,
                              },
                              "numberOfMonths": 2,
                              "onDayClick": [Function],
                              "selectedDays": Array [
                                undefined,
                                Object {
                                  "from": undefined,
                                  "to": undefined,
                                },
                              ],
                              "toMonth": undefined,
                            }
                          }
                          format="DD/MM/YY"
                          formatDate={[Function]}
                          hideOnDayClick={true}
                          inputProps={Object {}}
                          keepFocus={true}
                          onDayChange={[Function]}
                          overlayComponent={[Function]}
                          parseDate={[Function]}
                          placeholder="dd/mm/yy"
                          showOverlay={false}
                          typedValue=""
                          value=""
                        >
                          <div
                            className="DayPickerInput"
                          >
                            <input
                              onBlur={[Function]}
                              onChange={[Function]}
                              onClick={[Function]}
                              onFocus={[Function]}
                              onKeyDown={[Function]}
                              onKeyUp={[Function]}
                              placeholder="dd/mm/yy"
                              value=""
                            />
                          </div>
                        </DayPickerInput>
                      </div>
                      <div
                        className="input-from-to input-from-to--to"
                      >
                        <label />
                        <DayPickerInput
                          classNames={
                            Object {
                              "container": "DayPickerInput",
                              "overlay": "DayPickerInput-Overlay",
                              "overlayWrapper": "DayPickerInput-OverlayWrapper",
                            }
                          }
                          clickUnselectsDay={false}
                          component="input"
                          dayPickerProps={
                            Object {
                              "disabledDays": Object {
                                "before": undefined,
                              },
                              "fromMonth": undefined,
                              "locale": "en",
                              "localeUtils": Object {
                                "formatDate": [Function],
                                "formatDay": [Function],
                                "formatMonthTitle": [Function],
                                "formatWeekdayLong": [Function],
                                "formatWeekdayShort": [Function],
                                "getFirstDayOfWeek": [Function],
                                "getMonths": [Function],
                                "parseDate": [Function],
                              },
                              "modifiers": Object {
                                "end": undefined,
                                "start": undefined,
                              },
                              "month": undefined,
                              "numberOfMonths": 2,
                              "selectedDays": Array [
                                undefined,
                                Object {
                                  "from": undefined,
                                  "to": undefined,
                                },
                              ],
                            }
                          }
                          format="DD/MM/YY"
                          formatDate={[Function]}
                          hideOnDayClick={true}
                          inputProps={Object {}}
                          keepFocus={true}
                          onDayChange={[Function]}
                          overlayComponent={[Function]}
                          parseDate={[Function]}
                          placeholder="dd/mm/yy"
                          showOverlay={false}
                          typedValue=""
                          value=""
                        >
                          <div
                            className="DayPickerInput"
                          >
                            <input
                              onBlur={[Function]}
                              onChange={[Function]}
                              onClick={[Function]}
                              onFocus={[Function]}
                              onKeyDown={[Function]}
                              onKeyUp={[Function]}
                              placeholder="dd/mm/yy"
                              value=""
                            />
                          </div>
                        </DayPickerInput>
                      </div>
                      <button
                        className="btn outline btn-remove"
                        onClick={[Function]}
                        type="button"
                      />
                    </div>
                  </RangeDayPicker>
                </div>
              </div>
            </Col>
            <Col
              className="mt-3 mt-xl-0"
              lg={12}
              md={12}
              sm={12}
              xl={5}
              xs={12}
            >
              <div
                className="mt-3 mt-xl-0 col-xl-5 col-lg-12 col-md-12 col-sm-12 col-12"
              >
                <Filter
                  changeFilter={[MockFunction]}
                  t={[Function]}
                >
                  <div
                    className="filter"
                  >
                    <FormControl
                      name="filter"
                      onChange={[Function]}
                      placeholder=""
                      type="text"
                      value=""
                    >
                      <input
                        className="form-control"
                        name="filter"
                        onChange={[Function]}
                        placeholder=""
                        type="text"
                        value=""
                      />
                    </FormControl>
                  </div>
                </Filter>
                <OrderBy
                  changeOrderBy={[MockFunction]}
                  t={[Function]}
                >
                  <div
                    className="order-by outline"
                  >
                    <FieldSwitch
                      classNames="icon-right"
                      handleChecked={[Function]}
                      icon="angle"
                      id="orderBy"
                      labelChecked=""
                      labelUnChecked=""
                      type="toggle-icon"
                    >
                      <div
                        className="field-checkbox"
                      >
                        <div
                          className="pretty custom p-icon p-toggle p-plain icon-right"
                        >
                          <input
                            checked={false}
                            id="orderBy"
                            onChange={[Function]}
                            type="checkbox"
                          />
                          <div
                            className="state p-on undefined"
                          >
                            <label />
                            <FontAwesomeIcon
                              border={false}
                              className=""
                              fixedWidth={false}
                              flip={null}
                              icon={
                                Object {
                                  "icon": Array [
                                    320,
                                    512,
                                    Array [],
                                    "f106",
                                    "M177 159.7l136 136c9.4 9.4 9.4 24.6 0 33.9l-22.6 22.6c-9.4 9.4-24.6 9.4-33.9 0L160 255.9l-96.4 96.4c-9.4 9.4-24.6 9.4-33.9 0L7 329.7c-9.4-9.4-9.4-24.6 0-33.9l136-136c9.4-9.5 24.6-9.5 34-.1z",
                                  ],
                                  "iconName": "angle-up",
                                  "prefix": "fas",
                                }
                              }
                              inverse={false}
                              listItem={false}
                              mask={null}
                              pull={null}
                              pulse={false}
                              rotation={null}
                              size={null}
                              spin={false}
                              swapOpacity={false}
                              symbol={false}
                              title=""
                              transform={null}
                            >
                              <svg
                                aria-hidden="true"
                                className="svg-inline--fa fa-angle-up fa-w-10 "
                                data-icon="angle-up"
                                data-prefix="fas"
                                focusable="false"
                                role="img"
                                style={Object {}}
                                viewBox="0 0 320 512"
                                xmlns="http://www.w3.org/2000/svg"
                              >
                                <path
                                  d="M177 159.7l136 136c9.4 9.4 9.4 24.6 0 33.9l-22.6 22.6c-9.4 9.4-24.6 9.4-33.9 0L160 255.9l-96.4 96.4c-9.4 9.4-24.6 9.4-33.9 0L7 329.7c-9.4-9.4-9.4-24.6 0-33.9l136-136c9.4-9.5 24.6-9.5 34-.1z"
                                  fill="currentColor"
                                  style={Object {}}
                                />
                              </svg>
                            </FontAwesomeIcon>
                          </div>
                          <div
                            className="state p-off"
                          >
                            <label />
                            <FontAwesomeIcon
                              border={false}
                              className=""
                              fixedWidth={false}
                              flip={null}
                              icon={
                                Object {
                                  "icon": Array [
                                    320,
                                    512,
                                    Array [],
                                    "f107",
                                    "M143 352.3L7 216.3c-9.4-9.4-9.4-24.6 0-33.9l22.6-22.6c9.4-9.4 24.6-9.4 33.9 0l96.4 96.4 96.4-96.4c9.4-9.4 24.6-9.4 33.9 0l22.6 22.6c9.4 9.4 9.4 24.6 0 33.9l-136 136c-9.2 9.4-24.4 9.4-33.8 0z",
                                  ],
                                  "iconName": "angle-down",
                                  "prefix": "fas",
                                }
                              }
                              inverse={false}
                              listItem={false}
                              mask={null}
                              pull={null}
                              pulse={false}
                              rotation={null}
                              size={null}
                              spin={false}
                              swapOpacity={false}
                              symbol={false}
                              title=""
                              transform={null}
                            >
                              <svg
                                aria-hidden="true"
                                className="svg-inline--fa fa-angle-down fa-w-10 "
                                data-icon="angle-down"
                                data-prefix="fas"
                                focusable="false"
                                role="img"
                                style={Object {}}
                                viewBox="0 0 320 512"
                                xmlns="http://www.w3.org/2000/svg"
                              >
                                <path
                                  d="M143 352.3L7 216.3c-9.4-9.4-9.4-24.6 0-33.9l22.6-22.6c9.4-9.4 24.6-9.4 33.9 0l96.4 96.4 96.4-96.4c9.4-9.4 24.6-9.4 33.9 0l22.6 22.6c9.4 9.4 9.4 24.6 0 33.9l-136 136c-9.2 9.4-24.4 9.4-33.8 0z"
                                  fill="currentColor"
                                  style={Object {}}
                                />
                              </svg>
                            </FontAwesomeIcon>
                          </div>
                        </div>
                      </div>
                    </FieldSwitch>
                  </div>
                </OrderBy>
              </div>
            </Col>
          </div>
        </Row>
      </div>
    </FilterRowsBlock>
  </div>
</CreateFilterRowBlockWrapper>
`;

exports[`Filter Row Block Component Renders correctly Mobile Version 1`] = `
<CreateFilterRowBlockWrapper>
  <div>
    <FilterRowsBlock
      changeFilterDateType={[MockFunction]}
      filterDateType="created"
      handleDateFrom={[MockFunction]}
      handleDateTo={[MockFunction]}
      handleOnChangeFilter={[MockFunction]}
      handleOnChangeOrderBy={[MockFunction]}
      handleResetDate={[MockFunction]}
      t={
        [MockFunction] {
          "calls": Array [
            Array [
              "filter.date.created",
            ],
            Array [
              "filter.date.updated",
            ],
            Array [
              "filter.date.created",
            ],
            Array [
              "filter.date.updated",
            ],
          ],
          "results": Array [
            Object {
              "type": "return",
              "value": "filter.date.created",
            },
            Object {
              "type": "return",
              "value": "filter.date.updated",
            },
            Object {
              "type": "return",
              "value": "filter.date.created",
            },
            Object {
              "type": "return",
              "value": "filter.date.updated",
            },
          ],
        }
      }
    >
      <div>
        <Row
          noGutters={false}
        >
          <div
            className="row"
          >
            <Col
              lg={12}
              md={12}
              sm={12}
              xl={7}
              xs={12}
            >
              <div
                className="col-xl-7 col-lg-12 col-md-12 col-sm-12 col-12"
              >
                <div
                  className="data-filter-by-date"
                >
                  <div
                    className="filter-date d-inline"
                  >
                    <div
                      className="pretty p-default p-round"
                      data-name="filter-date-created"
                    >
                      <input
                        checked={true}
                        name="filterDateType"
                        onChange={[Function]}
                        type="radio"
                        value="created"
                      />
                      <div
                        className="state p-info-o"
                      >
                        <label>
                          filter.date.created
                        </label>
                      </div>
                    </div>
                    <div
                      className="pretty p-default p-round"
                      data-name="filter-date-modified"
                    >
                      <input
                        checked={false}
                        name="filterDateType"
                        onChange={[Function]}
                        type="radio"
                        value="modified"
                      />
                      <div
                        className="state p-info-o"
                      >
                        <label>
                          filter.date.updated
                        </label>
                      </div>
                    </div>
                  </div>
                  <RangeDayPicker
                    dateFrom={[Function]}
                    dateTo={[Function]}
                    resetDate={[Function]}
                    t={[Function]}
                  >
                    <div
                      className="range-day-picker"
                    >
                      <div
                        className="input-from-to input-from-to--from"
                      >
                        <label />
                        <DayPickerInput
                          classNames={
                            Object {
                              "container": "DayPickerInput",
                              "overlay": "DayPickerInput-Overlay",
                              "overlayWrapper": "DayPickerInput-OverlayWrapper",
                            }
                          }
                          clickUnselectsDay={false}
                          component="input"
                          dayPickerProps={
                            Object {
                              "disabledDays": Object {
                                "after": undefined,
                              },
                              "locale": "en",
                              "localeUtils": Object {
                                "formatDate": [Function],
                                "formatDay": [Function],
                                "formatMonthTitle": [Function],
                                "formatWeekdayLong": [Function],
                                "formatWeekdayShort": [Function],
                                "getFirstDayOfWeek": [Function],
                                "getMonths": [Function],
                                "parseDate": [Function],
                              },
                              "modifiers": Object {
                                "end": undefined,
                                "start": undefined,
                              },
                              "numberOfMonths": 2,
                              "onDayClick": [Function],
                              "selectedDays": Array [
                                undefined,
                                Object {
                                  "from": undefined,
                                  "to": undefined,
                                },
                              ],
                              "toMonth": undefined,
                            }
                          }
                          format="DD/MM/YY"
                          formatDate={[Function]}
                          hideOnDayClick={true}
                          inputProps={Object {}}
                          keepFocus={true}
                          onDayChange={[Function]}
                          overlayComponent={[Function]}
                          parseDate={[Function]}
                          placeholder="dd/mm/yy"
                          showOverlay={false}
                          typedValue=""
                          value=""
                        >
                          <div
                            className="DayPickerInput"
                          >
                            <input
                              onBlur={[Function]}
                              onChange={[Function]}
                              onClick={[Function]}
                              onFocus={[Function]}
                              onKeyDown={[Function]}
                              onKeyUp={[Function]}
                              placeholder="dd/mm/yy"
                              value=""
                            />
                          </div>
                        </DayPickerInput>
                      </div>
                      <div
                        className="input-from-to input-from-to--to"
                      >
                        <label />
                        <DayPickerInput
                          classNames={
                            Object {
                              "container": "DayPickerInput",
                              "overlay": "DayPickerInput-Overlay",
                              "overlayWrapper": "DayPickerInput-OverlayWrapper",
                            }
                          }
                          clickUnselectsDay={false}
                          component="input"
                          dayPickerProps={
                            Object {
                              "disabledDays": Object {
                                "before": undefined,
                              },
                              "fromMonth": undefined,
                              "locale": "en",
                              "localeUtils": Object {
                                "formatDate": [Function],
                                "formatDay": [Function],
                                "formatMonthTitle": [Function],
                                "formatWeekdayLong": [Function],
                                "formatWeekdayShort": [Function],
                                "getFirstDayOfWeek": [Function],
                                "getMonths": [Function],
                                "parseDate": [Function],
                              },
                              "modifiers": Object {
                                "end": undefined,
                                "start": undefined,
                              },
                              "month": undefined,
                              "numberOfMonths": 2,
                              "selectedDays": Array [
                                undefined,
                                Object {
                                  "from": undefined,
                                  "to": undefined,
                                },
                              ],
                            }
                          }
                          format="DD/MM/YY"
                          formatDate={[Function]}
                          hideOnDayClick={true}
                          inputProps={Object {}}
                          keepFocus={true}
                          onDayChange={[Function]}
                          overlayComponent={[Function]}
                          parseDate={[Function]}
                          placeholder="dd/mm/yy"
                          showOverlay={false}
                          typedValue=""
                          value=""
                        >
                          <div
                            className="DayPickerInput"
                          >
                            <input
                              onBlur={[Function]}
                              onChange={[Function]}
                              onClick={[Function]}
                              onFocus={[Function]}
                              onKeyDown={[Function]}
                              onKeyUp={[Function]}
                              placeholder="dd/mm/yy"
                              value=""
                            />
                          </div>
                        </DayPickerInput>
                      </div>
                      <button
                        className="btn outline btn-remove"
                        onClick={[Function]}
                        type="button"
                      />
                    </div>
                  </RangeDayPicker>
                </div>
              </div>
            </Col>
            <Col
              className="mt-3 mt-xl-0"
              lg={12}
              md={12}
              sm={12}
              xl={5}
              xs={12}
            >
              <div
                className="mt-3 mt-xl-0 col-xl-5 col-lg-12 col-md-12 col-sm-12 col-12"
              >
                <Filter
                  changeFilter={[MockFunction]}
                  t={[Function]}
                >
                  <div
                    className="filter"
                  >
                    <FormControl
                      name="filter"
                      onChange={[Function]}
                      placeholder=""
                      type="text"
                      value=""
                    >
                      <input
                        className="form-control"
                        name="filter"
                        onChange={[Function]}
                        placeholder=""
                        type="text"
                        value=""
                      />
                    </FormControl>
                  </div>
                </Filter>
                <OrderBy
                  changeOrderBy={[MockFunction]}
                  t={[Function]}
                >
                  <div
                    className="order-by outline"
                  >
                    <FieldSwitch
                      classNames="icon-right"
                      handleChecked={[Function]}
                      icon="angle"
                      id="orderBy"
                      labelChecked=""
                      labelUnChecked=""
                      type="toggle-icon"
                    >
                      <div
                        className="field-checkbox"
                      >
                        <div
                          className="pretty custom p-icon p-toggle p-plain icon-right"
                        >
                          <input
                            checked={false}
                            id="orderBy"
                            onChange={[Function]}
                            type="checkbox"
                          />
                          <div
                            className="state p-on undefined"
                          >
                            <label />
                            <FontAwesomeIcon
                              border={false}
                              className=""
                              fixedWidth={false}
                              flip={null}
                              icon={
                                Object {
                                  "icon": Array [
                                    320,
                                    512,
                                    Array [],
                                    "f106",
                                    "M177 159.7l136 136c9.4 9.4 9.4 24.6 0 33.9l-22.6 22.6c-9.4 9.4-24.6 9.4-33.9 0L160 255.9l-96.4 96.4c-9.4 9.4-24.6 9.4-33.9 0L7 329.7c-9.4-9.4-9.4-24.6 0-33.9l136-136c9.4-9.5 24.6-9.5 34-.1z",
                                  ],
                                  "iconName": "angle-up",
                                  "prefix": "fas",
                                }
                              }
                              inverse={false}
                              listItem={false}
                              mask={null}
                              pull={null}
                              pulse={false}
                              rotation={null}
                              size={null}
                              spin={false}
                              swapOpacity={false}
                              symbol={false}
                              title=""
                              transform={null}
                            >
                              <svg
                                aria-hidden="true"
                                className="svg-inline--fa fa-angle-up fa-w-10 "
                                data-icon="angle-up"
                                data-prefix="fas"
                                focusable="false"
                                role="img"
                                style={Object {}}
                                viewBox="0 0 320 512"
                                xmlns="http://www.w3.org/2000/svg"
                              >
                                <path
                                  d="M177 159.7l136 136c9.4 9.4 9.4 24.6 0 33.9l-22.6 22.6c-9.4 9.4-24.6 9.4-33.9 0L160 255.9l-96.4 96.4c-9.4 9.4-24.6 9.4-33.9 0L7 329.7c-9.4-9.4-9.4-24.6 0-33.9l136-136c9.4-9.5 24.6-9.5 34-.1z"
                                  fill="currentColor"
                                  style={Object {}}
                                />
                              </svg>
                            </FontAwesomeIcon>
                          </div>
                          <div
                            className="state p-off"
                          >
                            <label />
                            <FontAwesomeIcon
                              border={false}
                              className=""
                              fixedWidth={false}
                              flip={null}
                              icon={
                                Object {
                                  "icon": Array [
                                    320,
                                    512,
                                    Array [],
                                    "f107",
                                    "M143 352.3L7 216.3c-9.4-9.4-9.4-24.6 0-33.9l22.6-22.6c9.4-9.4 24.6-9.4 33.9 0l96.4 96.4 96.4-96.4c9.4-9.4 24.6-9.4 33.9 0l22.6 22.6c9.4 9.4 9.4 24.6 0 33.9l-136 136c-9.2 9.4-24.4 9.4-33.8 0z",
                                  ],
                                  "iconName": "angle-down",
                                  "prefix": "fas",
                                }
                              }
                              inverse={false}
                              listItem={false}
                              mask={null}
                              pull={null}
                              pulse={false}
                              rotation={null}
                              size={null}
                              spin={false}
                              swapOpacity={false}
                              symbol={false}
                              title=""
                              transform={null}
                            >
                              <svg
                                aria-hidden="true"
                                className="svg-inline--fa fa-angle-down fa-w-10 "
                                data-icon="angle-down"
                                data-prefix="fas"
                                focusable="false"
                                role="img"
                                style={Object {}}
                                viewBox="0 0 320 512"
                                xmlns="http://www.w3.org/2000/svg"
                              >
                                <path
                                  d="M143 352.3L7 216.3c-9.4-9.4-9.4-24.6 0-33.9l22.6-22.6c9.4-9.4 24.6-9.4 33.9 0l96.4 96.4 96.4-96.4c9.4-9.4 24.6-9.4 33.9 0l22.6 22.6c9.4 9.4 9.4 24.6 0 33.9l-136 136c-9.2 9.4-24.4 9.4-33.8 0z"
                                  fill="currentColor"
                                  style={Object {}}
                                />
                              </svg>
                            </FontAwesomeIcon>
                          </div>
                        </div>
                      </div>
                    </FieldSwitch>
                  </div>
                </OrderBy>
              </div>
            </Col>
          </div>
        </Row>
      </div>
    </FilterRowsBlock>
  </div>
</CreateFilterRowBlockWrapper>
`;
